zebra-rpc/src/methods/get_block_template_rpcs.rs

```diff
@@ -325,13 +336,35 @@ where
             // Convert into TransactionTemplates
             let mempool_txs = mempool_txs.iter().map(Into::into).collect();
 
-            let empty_string = String::from("");
+            // Calling state with `ChainInfo` request for relevant chain data
+            let request = ReadRequest::ChainInfo;
+            let response = state
+                .ready()
+                .and_then(|service| service.call(request))
+                .await
+                .map_err(|error| Error {
+                    code: ErrorCode::ServerError(0),
+                    message: error.to_string(),
+                    data: None,
+                })?;
+
+            let chain_info = match response {
+                ReadResponse::ChainInfo(chain_info) => chain_info,
+                _ => unreachable!("chaininfo always respond even if all data inside is `None`"),
+            };
+
+            let expected_difficulty = chain_info.expected_difficulty.ok_or(Error {
+                code: ErrorCode::ServerError(0),
+                message: "Not enough blocks in the chain".to_string(),
+                data: None,
+            })?;
+
             Ok(GetBlockTemplate {
-                capabilities: vec![],
+                capabilities: vec!["proposal".to_string()],
```

Comment:
Zebra does not support the `proposal` feature, so the `capabilities` must be empty:
https://en.bitcoin.it/wiki/BIP_0022#Block_Template_Request
https://en.bitcoin.it/wiki/BIP_0023#Block_Proposal

We can open an optional ticket for it - we don't know if any mining pools need it yet.

```suggestion
                capabilities: Vec::new(),
```

Reply:
I don't understand this, zcash has this value hardcoded i dont see why we can't do the same. What is the worst thing that can happen if we just leave it as it is now ?

Reply:
> The block data MUST be validated and checked against the server's usual acceptance rules (excluding the check for a valid proof-of-work).

I suspect zcashd does support this (it was mentioned during an arborist call about zip-317 that miners could always include their own transactions and I've been wondering about how that could be reliable since then.)

We should consider supporting this too but a false positive might be jarring.

Reply:
If mining pools get this capability, they might submit blocks using `getblocktemplate` rather than `submitblock`.

We decided that we don't need a ticket to support this feature, because we don't know if any mining pools use it.